### ISSUE TITLE
Addslashes results in redundant database queries

### ISSUE THREAD

**Issue by jcushman** [dhepper/django-model-path-converter#3]
I tested this under Django 2.2, but believe the same issue would apply on master. If both of these things are true:

* you have the default `APPEND_SLASH = True`
* you visit a valid url that doesn't end in a slash

then Django ends up calling `ModelConverterMixin.to_python()` twice while processing the request, which results in a wasteful second database query for the same value. 

It gets called once in `CommonMiddleware.should_redirect_with_slash()`, when checking if the url without a slash is valid, and then a second time when actually routing the url.

I'm not sure if this is avoidable without a change to Django itself, and it's a bit of an edge case because most valid urls will probably end in slashes when APPEND_SLASH is True. Just wanted to flag it in case it affects anyone else.

**Comment by dhepper** [comment]
You are absolutely right. Unfortunately still an issue with Django 4.0 and I don't see a way to mitigate it as the interface for path converters is quite limited.
